designs/2021-suppression-support/README.md

```diff
@@ -0,0 +1,504 @@
+- Repo: eslint/eslint
+- Start Date: 2021-07-28
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Yiwei Ding
+
+# Add suppression information in the output of ESLint
+
+## Summary
+
+Violations can be suppressed by inline directive comments and external configuration files. We propose to add an `--track-suppressions` CLI option for adding suppression information, including its kind and justification, in the output of ESLint.
```

Comment:
i don't think the word "suppressed" or "suppression" is appropriate here. inline comments can also *enable* rules, or reconfigure them.

What happens with this CLI option when an inline comment causes an error to appear that wouldn't have otherwise done so?

Reply:
We are only tracking when a rule is disabled for this RFC. Enabling a rule with a directive doesn’t change how this behaves.

Is there a better word you can suggest?

Reply:
So If i *reconfigure* an enabled rule such that it provides fewer errors, that would not be caught by this rule, despite that being conceptually the same kind of "suppression" as disabling a rule entirely?

Reply:
Correct. We care only about rules that are completely turned off

Reply:
Then this isn't about warning suppression at all, it's just about disabled rules (which, incidentally, seems much much less useful than caring about "a comment that reduced the number of observed warnings).

Perhaps `--track-inline-disabled-rules`, since that's all it's doing?

Reply:
It’s also tracking rules disabled in configs, not just inline. 

And really, it’s not tracking that the rule is disabled but rather the messages that would have been produced by a disabled rule. “Suppression” refers to the message being squashed rather than the rule being disabled.

Reply:
Good to know it’s not just inline - in that case it’s only about disabled rules - meaning, eslint somehow tracks which rules are configured to be on, with an “off” trampling it later?

`--track-disabled-rules` still seems like a better option to me in that case.

Reply:
> eslint somehow tracks which rules are configured to be on, with an “off” trampling it later?

You may want to read the rest of the RFC.  😄 

I just don’t think “track disabled rules” is an accurate phrase for what this is doing.

Reply:
I have, altho it’s long and i may have missed something, but what i read implies that disabling a rule in a config file isn’t something eslint tracks. Is this rfc attempting to change that?

Reply:
We might have to change the logic of skipping rules disabled in configuration files, as described in "Reserve external suppression" section of the RFC.